iOS/DoraemonKit/Src/Core/Plugin/Performance/CPU/Function/DoraemonCPUUtil.m

```diff
@@ -24,23 +30,32 @@ + (CGFloat)cpuUsageForApp {
     //  获取当前进程中 线程列表
     kr = task_threads(mach_task_self(), &thread_list, &thread_count);
     if (kr != KERN_SUCCESS)
-        return -1;
+        return nil;
 
-    float tot_cpu = 0;
+    DoraemonCPUUsageCallStackModel *usageModel = [[DoraemonCPUUsageCallStackModel alloc] init];
     
     for (int j = 0; j < thread_count; j++) {
         thread_info_count = THREAD_INFO_MAX;
         //获取每一个线程信息
         kr = thread_info(thread_list[j], THREAD_BASIC_INFO,
                          (thread_info_t)thinfo, &thread_info_count);
         if (kr != KERN_SUCCESS)
-            return -1;
+            return nil;
         
         basic_info_th = (thread_basic_info_t)thinfo;
         if (!(basic_info_th->flags & TH_FLAGS_IDLE)) {
             // cpu_usage : Scaled cpu usage percentage. The scale factor is TH_USAGE_SCALE.
             //宏定义TH_USAGE_SCALE返回CPU处理总频率：
-            tot_cpu += basic_info_th->cpu_usage / (float)TH_USAGE_SCALE;
+            float usage = basic_info_th->cpu_usage / (float)TH_USAGE_SCALE;
+            usageModel.totalUsage += usage;
+            
+            if (usage > usageModel.maxUsageOfThread) {
+                NSString *callstack = [DoraemonBacktraceLogger doraemon_backtraceOfThread:thread_list[j]];
+                if (callstack.length > 0) {
+                    usageModel.maxUsageOfThread = usage;
```

Comment:
- the callstack by DoraemonBacktraceLogger  is no symbolization
- we give data to react,  the callStack don't need the all callstack,  maybe can give a func to get the main callstack

Reply:
* Any suggestions on getting symbolized call stack?
* I think it's out of the task, maybe we should have a discussion on it.